test/functional/apps/discover/group1/_no_data.ts

```diff
@@ -55,16 +61,20 @@ export default function ({ getService, getPageObjects }: FtrProviderContext) {
 
       const button = await testSubjects.find('createDataViewButton');
       button.click();
+      log.error('Create dataview button clicked');
       await retry.waitForWithTimeout('data view editor form to be visible', 15000, async () => {
         return await (await find.byClassName('indexPatternEditor__form')).isDisplayed();
       });
 
+      log.error('indexPatternEditor__form is displayed');
       const dataViewToCreate = 'logstash';
       await createDataView(dataViewToCreate);
+      log.error('data view created');
       await PageObjects.header.waitUntilLoadingHasFinished();
+      log.error('loading finished');
       await retry.waitForWithTimeout(
         'data view selector to include a newly created dataview',
-        5000,
+        10000,
```

Comment:
just for testing, and since you last change failed. For debugging it might make sense to add a `await this.browser.refresh();` for the case the currently selected data view can't be found and Discover's main view is not being displayed. (it doesn't fail locally for me, so I'm also wondering)

Reply:
the guess is, the data view was created, but when returning the api still return there's no data view, because it has not been index, oder there has been a cached response ... something like this